crates/api_models/src/payment_methods.rs

```diff
@@ -546,11 +546,250 @@ pub struct TokenizedCardValue2 {
 
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 pub struct TokenizedWalletValue1 {
-    pub issuer: String,
-    pub token: Option<String>,
+    pub data: WalletData,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub enum WalletData {
+    GpayWallet(GpayWalletData),
+    ApplePayWallet(ApplePayWalletData),
+    PayPalWallet(PayPalWalletData),
+    Paypal(PaypalRedirection),
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct PaypalRedirection {}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct GpayWalletData {
+    pub pm_type: String,
+    pub description: String,
+    pub info: GpayPaymentMethodInfo,
+    pub tokenization_data: GpayTokenizationData,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct GpayPaymentMethodInfo {
+    pub card_network: String,
+    pub card_details: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct GpayTokenizationData {
+    pub token_type: String,
+    pub token: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct ApplePayWalletData {
+    pub payment_data: ApplepayPaymentData,
+    pub payment_method: ApplepayPaymentMethod,
+    pub transaction_identifier: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct ApplepayPaymentData {
+    pub data: String,
+    pub signature: String,
+    pub header: ApplepayHeader,
+    pub version: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct ApplepayHeader {
+    pub public_key_hash: String,
+    pub ephemeral_public_key: String,
+    pub transaction_id: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct ApplepayPaymentMethod {
+    pub display_name: String,
+    pub network: String,
+    #[serde(rename = "type")]
+    pub pm_type: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct PayPalWalletData {
+    pub token: String,
 }
 
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 pub struct TokenizedWalletValue2 {
     pub customer_id: Option<String>,
 }
+
+impl From<payments::WalletData> for WalletData {
```

Comment:
Why we needed this conversion?

Reply:
For the data in TokenizedWalletValue1, it requires Walletdata of payment methods but we have wallet data of payments, so we need this conversion

Reply:
Used the same walletdata